components/StemPlayer.tsx

```diff
@@ -78,15 +78,33 @@ type StemPlayerProps = {
 	onSolo?: (idx: number) => any
 	onSkipPrev?: () => any
 	onStop?: () => any
+	onNewFile: (newFile: Blob) => void
 }
 
 const StemPlayer = (props: StemPlayerProps): JSX.Element => {
-	const { idx, details, onWavesInit, onFinish, isStemDetails, onSolo, onSkipPrev, onStop } = props
+	const { idx, details, onWavesInit, onFinish, isStemDetails, onSolo, onSkipPrev, onStop, onNewFile } = props
 	const [wavesurfer, setWavesurfer] = useState<WaveSurfer>()
 	const [isMuted, setIsMuted] = useState<boolean>(false)
 	const [isSoloed, setIsSoloed] = useState<boolean>(false)
+	const [blob, setBlob] = useState<Blob>()
+	const [loadingBlob, setLoadingBlob] = useState<boolean>(false)
 
 	useEffect(() => {
+		let blobFileToLoad: Blob | undefined
+		if (!blob) {
+			if (!loadingBlob) {
+				setLoadingBlob(true)
+				fetch(details.audioHref).then(resp => {
+					resp.blob().then(b => {
+						setBlob(b)
+						onNewFile(b)
+						blobFileToLoad = b
```

Comment:
Needed to set the blob to a ref while we have access to it within this then block, so we can load it later on outside of this scope.